src/Parallel/PupStlCpp17.hpp

```diff
@@ -20,9 +20,8 @@ void pup(er& p, std::optional<T>& t) noexcept {  // NOLINT
   if (p.isUnpacking()) {
     p | valid;
     if (valid) {
-      T value{};
-      p | value;
-      t = std::move(value);
```

Comment:
It's a little surprising to me that this fails when a class defines move assignment and not copy assignment. Do you understand why? Is std::optional not forwarding as it should?

Reply:
It's because the resulting object in FixedHashMap is a `std::pair<const Key, unique_ptr>` and the `const Key` makes it non-movable while the `unique_ptr` makes it non-copyable. I'll add a code comment and reword the commit message.